### PR TITLE
creamed component

### PR DESCRIPTION
**Pull request by spookydonut**
## About The Pull Request
I have no mouth but I must scream

## Why It's Good For The Game
creampie.dmi

## Changelog
:cl:
code: cream pie component makes use of some of the unused pieface sprites 
/:cl:

### CODE REVIEW

**Comment by optimumtact** [comment]
absolutely based and component pilled beyond belief

**Comment by Fikou** [comment]
can you make the unused creampie images for xenos borgs and monkeys not be unused thanks

**Comment by spookydonut** [comment]
> 
> 
> can you make the unused creampie images for xenos borgs and monkeys not be unused thanks

the what

**Comment by ExcessiveUseOfCobblestone** [comment]
stop

File: code/__DEFINES/components.dm [hunk 1]
```diff
@@ -283,6 +283,9 @@
 #define COMSIG_TURF_MAKE_DRY "make_turf_try"						//(max_strength, immediate, duration_decrease = INFINITY): Returns bool.
 #define COMSIG_COMPONENT_CLEAN_ACT "clean_act"					//called on an object to clean it of cleanables. Usualy with soap: (num/strength)
 
+//Creamed
+#define COMSIG_COMPONENT_CLEAN_FACE_ACT "clean_face_act"		//called when you wash your face
```
Comment (Qustinnus) [inline code/__DEFINES/components.dm]: Add params

Comment (Qustinnus) [reply code/__DEFINES/components.dm]: And source

**Review by ghost — APPROVED** [review]
very good use of components. Good job.

File: code/datums/components/creamed.dm [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+			creamface.icon_state = "creampie_monkey"
+		else if(iscorgi(parent))
+			creamface.icon_state = "creampie_corgi"
+		else if(isAI(parent))
+			creamface.icon_state = "creampie_ai"
+
+		var/atom/A = parent
+		A.add_overlay(creamface)
+
+	pie_hit(source, stunning)
```
Comment (ninjanomnom) [inline code/datums/components/creamed.dm]: This component has too many functions. The effects of the pie hitting shouldnt be tied to the graphics for it in component code.

File: code/datums/components/creamed.dm [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+/datum/component/creamed/InheritComponent(datum/component/C, i_am_original, datum/source, stunning)
+	pie_hit(source, stunning)
```
Comment (ninjanomnom) [inline code/datums/components/creamed.dm]: No, the args are passed as a list, not individual arguments

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Pull request merged**